src/net_processing.cpp

```diff
@@ -2889,7 +2889,7 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
 bool PeerManagerImpl::ProcessOrphanTx(Peer& peer)
 {
     AssertLockHeld(g_msgproc_mutex);
-    AssertLockHeld(cs_main);
```

Comment:
Should become AssertLockNotHeld? :)

Reply:
Seems more reasonable to continue to allow invoking `ProcessOrphanTx` with `cs_main` held, even though we could forbid it. (Before the "only process orphans before messages" commit, `ProcessOrphanTx()` was called from `ProcessMessage()` with `cs_main` held)